lib/src/main/java/nl/knaw/dans/lib/dataverse/DatasetApi.java

```diff
@@ -190,6 +190,18 @@ public DataverseResponse<DatasetVersion> editMetadata(String s, Boolean replace)
         return putToTarget("editMetadata", s, queryParams, DatasetVersion.class);
     }
 
+    public DataverseResponse<DatasetVersion> editMetadata(String s, Boolean replace, HashMap<String, List<String>> extraQueryParams) throws IOException, DataverseException {
```

Comment:
Please, follow the pattern already established with the replace parameter. The library interface should present an appropriate level of abstraction. The caller of the library just wants to provide the metadataKey and should not be concerned with how it is sent. Also, it would be inconsistent to keep the "replace" argument separate.

Reply:
The name of the block and the value of the key (secret) must be passed. 
Also in general we might need to pass keys for several blocks. 

@janvanmansum So it would then become HashMap<String, String> metadataKeys. 
Internally converted to those query parameters.